### DISCUSSION TITLE
Is there a point to using HTTPS/SSL/TLS with Tor?

### DISCUSSION THREAD

**Discussion by yandodov** [globaleaks/globaleaks-whistleblowing-software#4157] [Q&A] [answered]
I've set up GlobaLeaks with HTTPS on my leaks.example.com domain. However, when I open the .onion link of GlobaLeaks in Brave with Tor, I can see the usual "not secure" message:

<img width="759" alt="image" src="https://github.com/user-attachments/assets/657895f3-8a98-49a0-aab2-ead1b927ab94">

This made me think — should I add HTTPS over Tor as well? I think that [this Stack Exchange answer](https://security.stackexchange.com/a/75984/256880) makes a lot of sense:

> When you are browsing TOR **hidden services**, the TOR system already provides end-to-end encryption, so **another TLS layer is redundant**. It is in fact counter-productive, because one of the goals of TLS is to de-anonymize the server. It makes no sense to go through the trouble to set up a hidden service and then get a certificate from a certificate authority which tells the whole world who you are.

Do you guys agree with this?

**Comment by evilaliv3** [comment, marked as answer]
Thank you for opening this discussion @hdodov 

The main advantage of adding HTTPS on Tor is to bind the onion address to the public domain thus improving the authority of the server. This of course make sense., but actually there are very few vendors that emit a certificate for .onion domains nor free certificates.
